pkg/instrumentors/bpf/github.com/gorilla/mux/probe.go

```diff
@@ -102,41 +97,52 @@ func (g *gorillaMuxInstrumentor) Load(ctx *context.InstrumentorContext) error {
 		return err
 	}
 
-	offset, err := ctx.TargetDetails.GetFunctionOffset(g.FuncNames()[0])
+	for _, funcName := range g.FuncNames() {
+		g.registerProbes(ctx, funcName)
+	}
+	rd, err := perf.NewReader(g.bpfObjects.Events, os.Getpagesize())
 	if err != nil {
 		return err
 	}
+	g.eventsReader = rd
+
+	return nil
+}
+
+func (g *gorillaMuxInstrumentor) registerProbes(ctx *context.InstrumentorContext, funcName string) {
+	logger := log.Logger.WithName("gorilla/mux-instrumentor").WithValues("function", funcName)
+	offset, err := ctx.TargetDetails.GetFunctionOffset(funcName)
+	if err != nil {
+		logger.Error(err, "could not find function start offset. Skipping")
+		return
+	}
+	retOffsets, err := ctx.TargetDetails.GetFunctionReturns(funcName)
+	if err != nil {
+		logger.Error(err, "could not find function end offset. Skipping")
```

Comment:
Can we include the funcName that failed in this logs?  Could be helpful for debugging in the future,

Reply:
The funcName is included on the log, as part of the logger being created a few lines up `logger := log.Logger.WithName("gorilla/mux-instrumentor").WithValues("function", funcName)`.

Actually this is making me realize we log errors from two places: once here in the `probe.go` and once in `analyze.go`.

I threw this error by adjusting line 114 to `offset, err := ctx.TargetDetails.GetFunctionOffset("test")`, so we see "error":"could not find offset for function **test**" where it would typically say "could not find offset for function github.com/gorilla/mux.(*Router).ServeHTTP"... 

```
{"level":"error","ts":1682102328.178772,"logger":"gorilla/mux-instrumentor","caller":"mux/probe.go:117","msg":"could not find function start offset. Skipping","function":"github.com/gorilla/mux.(*Router).ServeHTTP","error":"could not find offset for function test","stacktrace":"go.opentelemetry.io/auto/pkg/instrumentors/bpf/github.com/gorilla/mux.(*gorillaMuxInstrumentor).registerProbes\n\t/app/pkg/instrumentors/bpf/github.com/gorilla/mux/probe.go:117\ngo.opentelemetry.io/auto/pkg/instrumentors/bpf/github.com/gorilla/mux.(*gorillaMuxInstrumentor).Load\n\t/app/pkg/instrumentors/bpf/github.com/gorilla/mux/probe.go:101\ngo.opentelemetry.io/auto/pkg/instrumentors.(*instrumentorsManager).load\n\t/app/pkg/instrumentors/runner.go:84\ngo.opentelemetry.io/auto/pkg/instrumentors.(*instrumentorsManager).Run\n\t/app/pkg/instrumentors/runner.go:34\nmain.main\n\t/app/cli/main.go:86\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
```